deliver/lib/deliver/app_screenshot.rb

```diff
@@ -293,13 +302,27 @@ def self.devices
       }
     end
 
+    def self.resolve_ipadpro_conflict_if_needed(device_type, file_name)
+      is_3rd_gen = file_name.include?("(3rd generation)")
+      if is_3rd_gen
+        if device_type == ScreenSize::IOS_IPAD_PRO
+          return ScreenSize::IOS_IPAD_PRO129_3RD
+        elsif device_type == ScreenSize::IOS_IPAD_PRO_MESSAGES
+          return ScreenSize::IOS_IPAD_PRO129_3RD_MESSAGES
+        end
+      end
+      return device_type
+    end
+
     def self.calculate_screen_size(path)
```

Comment:
I think this method should return all available `device_types`(`[ScreenSize]`).

For example, image which is 2732x2048 is passed, this method will return `[ScreenSize::IOS_IPAD_PRO, IOS_IPAD_PRO129_3RD]`.
And filter this on out of this method.

Current implementation is difficult to scale for same situations in the future.

Reply:
It will introduce a lot of refactoring of this class (i.e `is_valid?` method, `initialize` now takes a single screen size as an argument). Are we sure we want to change that?

Reply:
FYI I'm currently trying to work out a new approach that will hopefully see this function removed in the future so maybe it doesn't matter for now? 🤔